WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -39,31 +47,43 @@ public AliceClient(RoundState roundState, ArenaClient arenaClient, SmartCoin coi
 		private long MaxVsizeAllocationPerAlice { get; }
 		private TimeSpan ConfirmationTimeout { get; }
 
-		public async Task RegisterAndConfirmInputAsync(RoundStateUpdater roundStatusUpdater, CancellationToken cancellationToken)
+		public static async Task<AliceClient> CreateRegisterAndConfirmInputAsync(
+			RoundState roundState,
+			ArenaClient arenaClient,
+			SmartCoin coin,
+			BitcoinSecret bitcoinSecret,
+			RoundStateUpdater roundStatusUpdater,
+			CancellationToken cancellationToken)
 		{
+			AliceClient? aliceClient = null;
 			try
 			{
-				await RegisterInputAsync(cancellationToken).ConfigureAwait(false);
-				await ConfirmConnectionAsync(roundStatusUpdater, cancellationToken).ConfigureAwait(false);
+				aliceClient = await RegisterInputAsync(roundState, arenaClient, coin, bitcoinSecret, cancellationToken).ConfigureAwait(false);
+				await aliceClient.ConfirmConnectionAsync(roundStatusUpdater, cancellationToken).ConfigureAwait(false);
 			}
 			catch (OperationCanceledException)
 			{
-				await TryToUnregisterAlicesAsync(CancellationToken.None).ConfigureAwait(false);
+				if (aliceClient is { })
+				{
+					await aliceClient.TryToUnregisterAlicesAsync(CancellationToken.None).ConfigureAwait(false);
+				}
+
 				throw;
 			}
+
+			return aliceClient;
 		}
 
-		private async Task RegisterInputAsync(CancellationToken cancellationToken)
+		private static async Task<AliceClient> RegisterInputAsync(RoundState roundState, ArenaClient arenaClient, SmartCoin coin, BitcoinSecret bitcoinSecret, CancellationToken cancellationToken)
 		{
+			AliceClient? aliceClient;
 			try
 			{
-				var response = await ArenaClient.RegisterInputAsync(RoundId, SmartCoin.Coin.Outpoint, BitcoinSecret.PrivateKey, cancellationToken).ConfigureAwait(false);
-                AliceId = response.Value;
-				SmartCoin.CoinJoinInProgress = true;
+				coin.CoinJoinInProgress = true;
```

Comment:
In this case you would need to switch it back to false when exception.

Reply:
ACK